js/app.js

```diff
@@ -0,0 +1,104 @@
+function createPerson4Request(url, id, propertyName) {
+  let oReq = new XMLHttpRequest();
+  oReq.addEventListener('load', changeWhat);
+  oReq.open('GET', url);
+  oReq.send();
+
+  function changeWhat() {
+    let response = JSON.parse(this.responseText);
+    console.log(response);
+    document.querySelector(id.concat(propertyName)).innerHTML = response.name;
+    changeWorld(response.homeworld);
+  }
+}
+
+function createPerson14Request(url, id, propertyName) {
+  let oReq = new XMLHttpRequest();
+  oReq.addEventListener('load', changeWhat);
+  oReq.open('GET', url);
+  oReq.send();
+
+  function changeWhat() {
+    let response = JSON.parse(this.responseText);
+    console.log(response);
+    document.querySelector(id.concat(propertyName)).innerHTML = response.name;
+    changeSpecies(response.species[0]);
+  }
+}
+
+function changeWorld(worldUrl) {
+  let worldReq = new XMLHttpRequest();
+  worldReq.addEventListener('load', setWorld);
+  worldReq.open('GET', worldUrl);
+  worldReq.send();
+
+  function setWorld() {
+    let worldResponse = JSON.parse(this.responseText);
+    console.log(worldResponse);
+    document.querySelector('#person4HomeWorld').innerHTML = worldResponse.name;
+  }
+}
```

Comment:
Nothing wrong, here but you best practice to not declare functions within functions. Can read more from comments here [(gene83#1)](https://github.com/gene83/the-async-force-ep-1/pull/1)